optimizer/validator.go

```diff
@@ -78,6 +80,10 @@ func (v *validator) Leave(in ast.Node) (out ast.Node, ok bool) {
 		}
 	case *ast.PatternInExpr:
 		v.checkSameColumns(append(x.List, x.Expr)...)
+	case *ast.Limit:
+		if x.Count > math.MaxUint64-x.Offset {
```

Comment:
Use the same style to check uint64 overflow?

+		limitCount := limit.Offset + limit.Count
 -		if limitCount < math.MaxInt64 {
 +			x.LimitCount = proto.Int64(int64(limitCount))
 +		}